### PR TITLE
S3

### PR DESCRIPTION
**Pull request by toonbie**

### CODE REVIEW

**Comment by toonbie** [comment]
If you close your eyes and imagine a happy client the story is perfect and needs no changes

File: src/component/ClientDropdown/index.jsx [hunk 1]
```diff
@@ -1,26 +1,15 @@
...
-    const [selectedClient, setSelectedClient] = useState('')
-
-    function storeClient(e) {
-        setSelectedClient(e.target.value)
-        console.log(clients[e.target.value])
-    }
-    
-    return(
-
+function ClientDropdown({ clients, selectedClient, setSelectedClient, storeClient }) {
```
Comment (ashleycoles) [inline src/component/ClientDropdown/index.jsx]: Are all of these props actually used?

File: src/component/CreateInvoice/index.jsx [hunk 2]
```diff
@@ -34,49 +36,101 @@ function CreateInvoice() {
...
+    function minusDetails() {
+        if (details.length > 1) {
+            setDetails(details.slice(0, -1))
+        }
     }
 
-    return (
-        <div className="bg-white">
+    function setRowDetails(newDetails, id) {
+        var newList = details.slice()
```
Comment (ashleycoles) [inline src/component/CreateInvoice/index.jsx]: use let instead

File: src/component/InvoiceRow/index.jsx [hunk 3]
```diff
@@ -12,21 +13,32 @@ function InvoiceRow({addDetails}) {
...
 
     }
-    
+    function handleDesc(e) {
+        setDesc(e.target.value)
+
+    }
+
+    useEffect(() => {
+        var newDetails = {
```
Comment (ashleycoles) [inline src/component/InvoiceRow/index.jsx]: lettttttttttt

File: src/component/InvoiceRow/index.jsx [hunk 4]
```diff
@@ -12,21 +13,32 @@ function InvoiceRow({addDetails}) {
         setRate(Number(e.target.value))
 
     }
-    
+    function handleDesc(e) {
+        setDesc(e.target.value)
+
```
Comment (ashleycoles) [inline src/component/InvoiceRow/index.jsx]: empty line can go

File: src/component/Header/index.jsx [hunk 5]
```diff
@@ -12,9 +13,14 @@ function Header() {
...
-        <header className="pt-8 inline-block">
-            <h1 className="font-bold text-4xl pb-1">Invoices</h1>
-            <p className="text-gray-500 pb-8 text-lg font-medium">There are {unpaid} unpaid invoices</p>
+        <header className="pt-8 flex justify-between">
+            <div>
+                <Link to="/"><h1 className="font-bold text-4xl pb-1">Invoices</h1></Link>
+                <p className="text-gray-500 pb-8 text-lg font-medium">There are {unpaid} unpaid invoices</p>
+            </div>
+            <div className="flex flex-col justify-center">
+                <Link className="bg-cyan-500 px-3 py-2 rounded text-white " to="/CreateInvoice"><i className="bg-black rounded-md px-1 not-italic">+</i> New Invoice</Link>
```
Comment (ashleycoles) [inline src/component/Header/index.jsx]: remove the empty space at the end of the className

File: src/component/CreateInvoice/index.jsx [hunk 6]
```diff
@@ -34,49 +36,101 @@ function CreateInvoice() {
...
+            <div className="bg-white pt-5 pb-5 pr-5 border-b flex justify-end">
+                {<p className="text-red-600 italic font-mono self-center animate-bounce">{errorMessage}</p>}
             </div>
-            {details.map(detail => <InvoiceRow addDetails={addDetails} />}
-            {console.log(details)}
-
-        </div>
+            <div className="bg-white pt-4 pb-10 px-3 flex justify-end gap-2 ">
+                <button onClick={sendInvoice} className="bg-green-600 text-white p-2 rounded">Create invoice</button>
+                <button className="bg-red-500 text-white p-2 rounded">Cancel invoice</button>
```
Comment (ashleycoles) [inline src/component/CreateInvoice/index.jsx]: Does this do anything?

File: src/component/CreateInvoice/index.jsx [hunk 7]
```diff
@@ -34,49 +36,101 @@ function CreateInvoice() {
...
                 </div>
             </div>
-
-            <div className="grid grid-cols-4 font-bold my-1 border-b border-t border-gray-300  ">
-                <p>Description</p>
-                <p>Quantity</p>
-                <p>Rate</p>
-                <p>Total</p>
+            <div className="bg-white pt-5 pb-5 pr-5 border-b flex justify-end">
+                {<p className="text-red-600 italic font-mono self-center animate-bounce">{errorMessage}</p>}
```
Comment (ashleycoles) [inline src/component/CreateInvoice/index.jsx]: Why is this whole thing in {}?

File: src/component/CreateInvoice/index.jsx [hunk 8]
```diff
@@ -34,49 +36,101 @@ function CreateInvoice() {
...
-                    <p className="font-medium pb-2 pt-2">From</p>
-                    <p>Kermit the Frog</p>
-                    <p>Beverly Hills</p>
-                    <p>California</p>
-                    <p className="font-medium pb-2 pt-4">To</p>
-                    <Dropdown clients={clients} />
+    function sendInvoice() {
+        const dataToSend = {
+            "client": selectedClient,
+            "total": details.reduce((carry, detail) => carry + detail.total, 0),
```
Comment (ashleycoles) [inline src/component/CreateInvoice/index.jsx]: You're having to calculate the total twice, once here and once in the JSX. Can you find a way to only need this once?

File: src/component/CreateInvoice/index.jsx [hunk 9]
```diff
@@ -3,14 +3,16 @@ import Dropdown from "../ClientDropdown"
...
 
 function CreateInvoice() {
+    const navigate = useNavigate()
     const [formattedDateCreated, setFormattedDateCreated] = useState("")
     const [formattedDateDue, setFormattedDateDue] = useState("")
 
-    const [clients, setClients] = useState([])
-
+    const [selectedClient, setSelectedClient] = useState("1")
+    const [clients, setClients] = useState([{ id: "" }])
```
Comment (ashleycoles) [inline src/component/CreateInvoice/index.jsx]: Why not just make this an empty array?

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: src/component/CreateInvoice/index.jsx (hunk 8 above)
Comment (toonbie) [reply src/component/CreateInvoice/index.jsx]: No, I cannot ; function over beauty

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**